Bio/Cluster/cluster.c

```diff
@@ -34,6 +34,14 @@
 #include <string.h>
 #include "cluster.h"
 
+#ifdef CLUSTER_USE_PYTHON_MEMORY
+#include "Python.h"
+#define malloc PyMem_Malloc
+#define calloc PyMem_Calloc
+#define realloc PyMem_Realloc
+#define free PyMem_Free
+#endif
+
```

Comment:
Can you change this to
```C
#ifdef CLUSTER_USE_PYTHON_MEMORY
#include "Python.h"
#define MALLOC PyMem_RawMalloc
#define CALLOC PyMem_RawCalloc
#define REALLOC PyMem_RawRealloc
#define FREE PyMem_RawFree
#else
#define MALLOC malloc
#define CALLOC calloc
#define REALLOC realloc
#define FREE free
#endif
```
as I am not sure if it is safe to just redefine malloc etc., and also the code should remain readable to people who do not use Python.
Then, you can just replace all instances of malloc/calloc/realloc/free by their equivalent in uppercase.

Reply:
Code looks good now, thanks!